loggie运行一段时间崩溃，提示panic: query registry fail: database is locked

报错信息如下，请问会是什么原因呢，我改如何排查呢

```
2023-04-12 21:15:14 INF data0/budong/loggie/pkg/source/file/watch.go:677 > watcher stop
panic: query registry fail: database is locked

goroutine 118 [running]:
github.com/loggie-io/loggie/pkg/util/persistence/driver.(*Engine).findBySql(0xc000cbe558, {0xc000ffe000, 0xc8})
        /data0/budong/loggie/pkg/util/persistence/driver/sqlite.go:263 +0x5da
github.com/loggie-io/loggie/pkg/util/persistence/driver.(*Engine).FindBy(0x30?, {0xc00061a3d0?, 0xc001000b60?}, {0xc000ba7198, 0x5}, {0xc000ba7168, 0x4})
        /data0/budong/loggie/pkg/util/persistence/driver/sqlite.go:162 +0x11b
github.com/loggie-io/loggie/pkg/util/persistence.(*DbHandler).FindBy(0x7f098ab8e000?, {0xc00061a3d0, 0xd}, {0xc000ba7198, 0x5}, {0xc000ba7168, 0x4})
        /data0/budong/loggie/pkg/util/persistence/persistence.go:189 +0xcd
github.com/loggie-io/loggie/pkg/source/file.(*Watcher).findExistJobRegistry(...)
        /data0/budong/loggie/pkg/source/file/watch.go:143
github.com/loggie-io/loggie/pkg/source/file.(*Watcher).eventBus(0xc000cee0c0, {0xc000a1ac90?, 0xc00033e280?, {0x0?, 0xc0014b0410?}})
        /data0/budong/loggie/pkg/source/file/watch.go:283 +0x1e5
github.com/loggie-io/loggie/pkg/source/file.(*Watcher).createOrRename(0xc000cee0c0, {0xc0045425d0?, 0x1d?}, 0xc0005decf8?, 0x0)
        /data0/budong/loggie/pkg/source/file/watch.go:444 +0x234
github.com/loggie-io/loggie/pkg/source/file.(*Watcher).scanPaths(0xc0004daa48?, {0xc000ba7168, 0x4}, {0xc000ba7198, 0x5}, {0xc000a08ab0?, 0x1, 0xc0005dee20?}, 0x4d6877?, 0x0)
        /data0/budong/loggie/pkg/source/file/watch.go:426 +0x30b
github.com/loggie-io/loggie/pkg/source/file.(*Watcher).scanTaskNewFiles(0x20?, 0x5?)
        /data0/budong/loggie/pkg/source/file/watch.go:370 +0x92
github.com/loggie-io/loggie/pkg/source/file.(*Watcher).scanNewFiles(...)
        /data0/budong/loggie/pkg/source/file/watch.go:357
github.com/loggie-io/loggie/pkg/source/file.(*Watcher).scan(0xc000cee0c0)
        /data0/budong/loggie/pkg/source/file/watch.go:524 +0x9e
github.com/loggie-io/loggie/pkg/source/file.(*Watcher).run(0xc000cee0c0)
        /data0/budong/loggie/pkg/source/file/watch.go:695 +0x2bb
created by github.com/loggie-io/loggie/pkg/source/file.newWatcher
        /data0/budong/loggie/pkg/source/file/watch.go:83 +0x2ac
```

--- a/pkg/util/persistence/driver/sqlite.go
+++ b/pkg/util/persistence/driver/sqlite.go
@@ -28,6 +28,7 @@ import (
 	"os"
 	"path/filepath"
 	"strings"
+	"sync"
 )
 
 const (
@@ -58,6 +59,8 @@ const (
 type Engine struct {
 	db     *sql.DB
 	dbFile string
+
+	mutex sync.Mutex
 }
 
 func Init(file string) *Engine {
@@ -78,6 +81,8 @@ func Init(file string) *Engine {
 }
 
 func (e *Engine) Insert(registries []reg.Registry) error {
+	e.mutex.Lock()
+	defer e.mutex.Unlock()
 	e.txWrapper(insertSql, func(stmt *sql.Stmt) {
 		for _, r := range registries {
 			_, err := stmt.Exec(r.PipelineName, r.SourceName, r.Filename, r.JobUid, r.Offset, r.CollectTime, r.Version, r.LineNumber)
@@ -94,6 +99,9 @@ func (e *Engine) Close() error {
 }
 
 func (e *Engine) Update(registries []reg.Registry) error {
+	e.mutex.Lock()
+	defer e.mutex.Unlock()
+
 	e.txWrapper(updateSql, func(stmt *sql.Stmt) {
 		for _, r := range registries {
 			_, err := stmt.Exec(r.Offset, r.CollectTime, r.LineNumber, r.Id)
@@ -107,6 +115,9 @@ func (e *Engine) Update(registries []reg.Registry) error {
 }
 
 func (e *Engine) UpdateFileName(rs []reg.Registry) error {
+	e.mutex.Lock()
+	defer e.mutex.Unlock()
+
 	e.txWrapper(updateNameByJobWatchId, func(stmt *sql.Stmt) {
 		for _, r := range rs {
 			result, err := stmt.Exec(r.Filename, r.JobUid, r.SourceName, r.PipelineName)
@@ -127,6 +138,9 @@ func (e *Engine) UpdateFileName(rs []reg.Registry) error {
 }
 
 func (e *Engine) Delete(r reg.Registry) error {
+	e.mutex.Lock()
+	defer e.mutex.Unlock()
+
 	var resErr error
 	e.txWrapper(deleteById, func(stmt *sql.Stmt) {
 		_, err := stmt.Exec(r.Id)
@@ -139,6 +153,9 @@ func (e *Engine) Delete(r reg.Registry) error {
 }
 
 func (e *Engine) DeleteBy(jobUid string, sourceName string, pipelineName string) error {
+	e.mutex.Lock()
+	defer e.mutex.Unlock()
+
 	var resErr error
 	e.txWrapper(deleteByJobWatchId, func(stmt *sql.Stmt) {
 		result, err := stmt.Exec(jobUid, sourceName, pipelineName)
@@ -225,6 +242,9 @@ func (e *Engine) txWrapper(sqlString string, f func(stmt *sql.Stmt)) {
 }
 
 func (e *Engine) findBySql(querySql string) []reg.Registry {
+	e.mutex.Lock()
+	defer e.mutex.Unlock()
+
 	rows, err := e.db.Query(querySql)
 	if err != nil {
 		panic(fmt.Sprintf("%s query registry by sql(%s) fail: %v", e.String(), querySql, err))

